.circleci/config.yml

```diff
@@ -6,55 +6,56 @@ jobs:
     working_directory: ~/repo
     steps:
       - checkout
+
       # Download and cache dependencies
       - restore_cache:
           keys:
-          - v1-dependencies-{{ checksum "package.json" }}
+          - v10-dependencies-{{ checksum "package.json" }}
           # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
+          - v10-dependencies-
       - run: npm install
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-{{ checksum "package.json" }}
+          key: v10-dependencies-{{ checksum "package.json" }}
 
+      - run: npm run build
       - run: npm test --coverage
       - run: npm run lint
       - run: npm run report-coverage
+      - persist_to_workspace:
+          root: ~/repo
+          paths: .
 
   publish_to_npm:
     docker:
       - image: circleci/node:10
     working_directory: ~/repo
     steps:
-      - checkout
-      # Download and cache dependencies
-      - restore_cache:
-          keys:
-          - v1-dependencies-{{ checksum "package.json" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
-
-      - run: git config --global user.email "info@circleci.com"
-      - run: git config --global user.name "CircleCI"
-      - run: git config --global push.default simple
-      - run: node scripts/publish_web.js
+      - attach_workspace:
+          at: ~/repo
       - setup_remote_docker
+      - run:
+          name: Authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc
+      - run:
+          name: Publish to NPM
+          command: npm publish
       - run: ./scripts/ci-deploy.sh
 
-
 workflows:
   version: 2
   build_and_publish:
     jobs:
       - build:
           filters:
-            branches:
-              ignore:
-                - gh-pages
+            tags:
+              only: /.*/
```

Comment:
What is this filter for?

Reply:
This was copied from the second example [here](https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag). Since the `publish_to_npm` job has tag filters and requires the `build` job, the `build` job must also match any tags the `publish_to_npm` job matches -- so I believe the filter matches any tag